src/data/proxy/DDBPreferenceCorrelationGraph.java

```diff
@@ -108,4 +115,18 @@ public Preference getPreference(String id, PreferenceCategory category) {
             return new DDBPreferenceAdapter(item).toObject();
         }
     }
+    
+    private List<Preference> getPreferences(Page<Item, ScanOutcome> page) {
+    	Iterator<Item> iterator = page.iterator();
+    	return Stream.generate(iterator::next).map(item -> new DDBPreferenceAdapter(item).toObject()).collect(Collectors.toList());
+    }
+
+	@Override
+	public Stream<List<Preference>> batchGetPreferences(
```

Comment:
This is kind of fundamental to the design.  batchGetPreference doesn't return all of the preferences at once, it just lazily returns batches of Preference(s) as they are requested.  Note it is not simply a List of all of the Preference(s), it is a Stream of List of Preferences, where each List is one batch.  I could have it return a List of Lists of Preference(s) but then you would be eagerly materializing all of the Preference(s) whereas the idea here is that we don't produce all of the Preference(s) at once but rather produce them on demand.  If we had a very large number of Preference(s) we could potentially run out of memory or take a very long time to produce them.